ios/browser/ui/webui/ai_chat/ai_chat_ui.mm

```diff
@@ -154,14 +155,15 @@
 
 void AIChatUI::BindInterfaceHistoryUIHandler(
     mojo::PendingReceiver<ai_chat::mojom::HistoryUIHandler> receiver) {
-  // TODO: https://github.com/brave/brave-browser/issues/51184 Add support
-  // for associating bookmarks/history
-  NOTIMPLEMENTED();
+  history_ui_handler_ = std::make_unique<ai_chat::HistoryUIHandler>(
+      std::move(receiver), ios::HistoryServiceFactory::GetForProfile(
+                               profile_, ServiceAccessType::EXPLICIT_ACCESS));
+  CHECK(history_ui_handler_);
 }
 
 void AIChatUI::BindInterfaceBookmarksPageHandler(
     mojo::PendingReceiver<ai_chat::mojom::BookmarksPageHandler> receiver) {
-  // TODO: https://github.com/brave/brave-browser/issues/51184 Add support
-  // for associating bookmarks/history
-  NOTIMPLEMENTED();
+  bookmarks_page_handler_ = std::make_unique<ai_chat::BookmarksPageHandler>(
+      ios::BookmarkModelFactory::GetForProfile(profile_), std::move(receiver));
```

Comment:
do we want to say its ExplicitAcesss here too?

Reply:
There's no `ServiceAccessType` for the `BookmarkModelFactory`